docs/IntegrationWithExistingApps.md

```diff
@@ -720,6 +720,8 @@ public boolean onKeyUp(int keyCode, KeyEvent event) {
 
 That's it, your activity is ready to run some JavaScript code.
 
+> If your app is targeting the Android `api level 23` or greater, make sure you have, for the development build, the `overlay permission` enabled. You can check it with `Settings.canDrawOverlays(this);`. This is required because, if your app produces an error in the react native component, the error view is displayed above all the other windows. Due to the new permissions system, introduced in the api level 23, the user needs to approve it.
```

Comment:
@satya164 Thanks a lot for your inputs. Very informational. I guess the only thing remaining for me is to figure out if I can somehow get rid of the `SYSTEM_ALERT_WINDOW` permission in the production apks. I will generate an apk after removing the permission from the `AndroidManifest.xml` file and check what kind of permissions are requested by the app when it is first installed. Like you said, the RN lib may be adding this permission even if I remove it from the manifest file which I can confirm by this test and if that is the case, I will try to figure out if I can somehow remove it in the production apks even if it means that I may have to make some manual changes before generating them. I guess my only concern is that I don't want to ask for permissions which I don't need and can't explain in the app's terms and conditions document.